Bluetooth device discovery agent (bluez): Prevent null pointer access.

We would potentially access a null adapter if stop() was called from a
user slot connected to the deviceDiscovered() signal.
Task-number: QTBUG-48481
Change-Id: I784120dcd7a9f71d9532fac9538ae98e1a329db2
Reviewed-by: Alex Blasche <alexander.blasche@theqtcompany.com>
diff -Nur qtconnectivity-opensource-src-5.5.1/src/bluetooth/qbluetoothdevicediscoveryagent_bluez.cpp qtconnectivity-opensource-src-5.5.1n/src/bluetooth/qbluetoothdevicediscoveryagent_bluez.cpp
--- qtconnectivity-opensource-src-5.5.1/src/bluetooth/qbluetoothdevicediscoveryagent_bluez.cpp	2015-10-13 12:35:37.000000000 +0800
+++ qtconnectivity-opensource-src-5.5.1n/src/bluetooth/qbluetoothdevicediscoveryagent_bluez.cpp	2015-11-07 11:09:00.487927356 +0800
@@ -247,6 +247,8 @@
                         continue; //devices whose path doesn't start with same path we skip
 
                     deviceFoundBluez5(path.path());
+                    if (!isActive()) // Can happen if stop() was called from a slot in user code.
+                        return;
                 }
             }
         }
